src/javascript/reducers/agreementList.js

```diff
@@ -23,135 +25,223 @@ type Action =
+
+  const allAgreementIds: Array<string> = state.allAgreementIds,
+        agreementsById: AgreementListType = state.agreementsById;
+
+  const sortedAgreements: Array<string> = allAgreementIds.sort((a, b) => {
+
+    const aAgreement: string = agreementsById[a][sortKey];
+    const bAgreement: string = agreementsById[b][sortKey];
+
+    if (sortKind === "ascending") {
+      if (aAgreement < bAgreement) return -1;
+      else if (aAgreement > bAgreement) return 1;
+      return 0;
+
+    } else if (sortKind === "descending") {
+      if (aAgreement > bAgreement) return -1;
+      else if (aAgreement < bAgreement) return 1;
+      return 0;
+    };
+    return 0;
+  });
+
+  return sortedAgreements;
+
 };
 
 export default function agreementListReducer(state: AgreementListState = INITIAL_STATE, action: Action): AgreementListState {
   // set agreementId, and check if payload passes an object with an error
-  // or passes just the agreement int. This could be a very long
-
+  // or passes just the agreement id
+  const agreementsById: AgreementListType = state.agreementsById;
   let agreementId: string = '';
+  let agreement: AgreementType;
 
-  // flow really doesn't like this ternary operator, makes me super sad
-  // if (action.payload) {
-  //   agreementId =  typeof action.payload === 'object' ?  action.payload.agreementId.toString(): action.payload.toString();
-  // }
-
-  if (typeof action.payload === 'object') {
+  // REALLY ugly way of making sure that agreementId is in the object, is there
+  // a better solution to this?
+  if (typeof action.payload === 'object' && action.payload.agreementId) {
+    //$FlowFixMe
     agreementId = action.payload.agreementId.toString();
-  } else if (action.payload) {
+  } else if (typeof action.payload === 'string') {
     agreementId = action.payload.toString();
   }
-
-  const agreement: AgreementType = state[agreementId];
+  // once agreementId is there, get the agreementId object
+  // so we can copy with the object spread operator
+  if (agreementId !== '') {
+    agreement = agreementsById[agreementId];
+  }
 
   switch (action.type) {
     case 'REQUEST_PENDING':
       return {
         ...state,
-        [agreementId]: {
-          ...agreement,
-          requestPending: true,
+        agreementsbyId: {
+          ...agreementsById,
+          [agreementId]: {
+            ...agreement,
+            requestPending: true,
+          }
         }
       }
     // send button
     case 'BUYER_SENT':
       return {
         ...state,
-        [agreementId]: {
-          ...agreement,
-          buyerPaid: true,
-          buyerPaidDate: Date.now(),
-          error: "",
-          requestPending: false,
-        },
+        agreementsbyId: {
+          ...agreementsById,
+          [agreementId]: {
+            ...agreement,
+            buyerPaid: true,
+            buyerPaidDate: Date.now(),
```

Comment:
Date.now() works for when the buyer has just now paid. But when you reload the page two days afterward, we'll need to fetch buyerPaidDate using web3 instead, since Date.now() will not be correct.

I'm guessing though that you used Date.now() just as a mostly-correct placeholder until a reducer exists that will populate the agreementList using data from web3